javalib/src/main/scala/java/util/UUID.scala

```diff
@@ -136,13 +136,26 @@ object UUID {
   private final val NameBased = 3
   private final val Random = 4
 
-  private lazy val rng = new Random() // TODO Use java.security.SecureRandom
+  // Typed as `Random` so that the IR typechecks when SecureRandom is not available
+  private lazy val csprng: Random = new java.security.SecureRandom()
+  private lazy val randomUUIDBuffer: Array[Byte] = new Array[Byte](16)
```

Comment:
Is it really faster to keep this buffer, rather than allocating it every time?

Reply:
From benchmarks I had done for the fround polyfill not long ago, yes, it's really faster not to reallocate a small array every time. See this commit:
https://github.com/scala-js/scala-js/commit/9d60658727a99d1a72a163720c3471172750e64c
I believe the same effects would be observable in this case.

Reply:
OK, fair enough :)